ssg/build_ovals.py

```diff
@@ -260,106 +260,165 @@ def _check_rule_id(oval_file_tree, rule_id):
     return False
 
 
-def checks(env_yaml, yaml_path, oval_version, oval_dirs, build_ovals_dir=None):
-    """
-    Concatenate all XML files in the oval directory, to create the document
-    body. Then concatenates this with all XML files in the guide directories,
-    preferring {{{ product }}}.xml to shared.xml.
+def _create_output_directory(directory_path=None):
+    if directory_path and not os.path.exists(directory_path):
+        os.makedirs(directory_path)
 
-    oval_dirs: list of directory with oval files (later has higher priority)
 
-    Return: The document body
-    """
+def _store_intermediate_files(rule_id, xml_content, directory_path=None):
+    if not directory_path:
+        return
+    output_file_name = rule_id + ".xml"
+    output_filepath = os.path.join(directory_path, output_file_name)
+    with open(output_filepath, "w") as f:
+        f.write(xml_content)
 
-    body = []
-    product = utils.required_key(env_yaml, "product")
-    included_checks_count = 0
-    reversed_dirs = oval_dirs[::-1]  # earlier directory has higher priority
-    already_loaded = dict()  # filename -> oval_version
+
+def _create_local_env_yaml(env_yaml, rule):
     local_env_yaml = dict()
     local_env_yaml.update(env_yaml)
+    local_env_yaml['rule_id'] = rule.id_
+    local_env_yaml['rule_title'] = rule.title
+    prodtypes = parse_prodtype(rule.prodtype)
+    local_env_yaml['products'] = prodtypes  # default is all
+    return local_env_yaml
+
+
+def _process_check(_path, local_env_yaml, build_ovals_dir, already_loaded):
+    rule_id = utils.required_key(local_env_yaml, "rule_id")
+    oval_version = utils.required_key(
+        local_env_yaml, "target_oval_version_str")
+    product = utils.required_key(local_env_yaml, "product")
+    # To be compatible with the later checks, use the rule_id
+    # (i.e., the value of _dir) to recreate the expected filename if
+    # this OVAL was in a rule directory.
+    filename = "%s.xml" % rule_id
+
+    xml_content = process_file_with_macros(_path, local_env_yaml)
+
+    if not _check_is_applicable_for_product(xml_content, product):
```

Comment:
Hello @jan-cerny , I see similar piece of code in the function _process_oval_file. Would it be possible to deduplicate some parts of those functions?